src/source/Crypto/Tls_mbedtls.c

```diff
@@ -4,6 +4,28 @@
 #define LOG_CLASS "TLS_mbedtls"
 #include "../Include_i.h"
 
+// Read and parse CA certificate
+PRIVATE_API STATUS readAndParseCACertificate(PTlsSession pTlsSession)
+{
+    ENTERS();
+    STATUS retStatus = STATUS_SUCCESS;
+    SIZE_T cert_len = 0;
+    PBYTE cert_buf = NULL;
+
+    CHK_STATUS(readFile(KVS_CA_CERT_PATH, FALSE, NULL, &cert_len));
+    CHK(cert_len > 0, STATUS_INVALID_CERT_PATH_LENGTH);
+    cert_buf = (PBYTE) MEMCALLOC(1, cert_len + 1);
+    CHK(cert_buf != NULL, STATUS_NOT_ENOUGH_MEMORY);
+    CHK_STATUS(readFile(KVS_CA_CERT_PATH, FALSE, cert_buf, &cert_len));
+    CHK(mbedtls_x509_crt_parse(&pTlsSession->cacert, cert_buf, cert_len) == 0, STATUS_INVALID_CA_CERT_PATH);
```

Comment:
Suggest to capture the return value of `mbedtls_x509_crt_parse` and in CleanUp, use `mbedtls_strerror` for additional debugging.